core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -1017,6 +1019,30 @@ public void setProperty(final String tableName, final String property, final Str
     }
   }
 
+  @Override
+  public void modifyProperties(String tableName, final Consumer<Map<String,String>> mapMutator)
+      throws AccumuloException, AccumuloSecurityException, IllegalArgumentException,
+      ConcurrentModificationException {
+    EXISTING_TABLE_NAME.validate(tableName);
+    checkArgument(mapMutator != null, "mapMutator is null");
+
+    final Map<String,String> properties = ThriftClientTypes.CLIENT.execute(context,
```

Comment:
With some changes, this code may be able to gracefully handle multiple clients concurrently modifying props.  Not completely sure, but think something like the following may work.


Create a thrift type to return version+props.

```java
class ThriftVersionedProperties {
   long version;
   Map<String, String> props;
}
```

The following code was written quickly, trying to show how we could pass a version back and forth over thrift and use that version to detect concurrent changes and retry.

```java
public void modifyProperties(String tableName, final Consumer<Map<String,String>> mapMutator)
      throws AccumuloException, AccumuloSecurityException, IllegalArgumentException,
      ConcurrentModificationException {

   while(true) {
      // make thrift call to get props+version
      ThriftVersionedProps tvp = client.getTableProperties(...);
      
      // mutate the properties
      mapMutator.accept(tvp.properties);

      try {
          // pass the modified props+version back to thrift call.  The thrift call will throw a thrift exception if the version has changed since we read props above.
          client.modifyTableProperties(...,tableName, tvp);
           break;
      } catch (PropertiesVersionChanged pvc) {
          log.debug("Properties changed since reading, retrying...");
          //TODO maybe sleep using retry object
      } catch (TableNotFoundException e) {
         throw new AccumuloException(e);
      }
   }
}
```

Reply:
Could this be done as a separate issue?

Reply:
I was going to suggest that..I think this could probably be a follow on PR as an improvement.

Reply:
Opened https://github.com/apache/accumulo/issues/2821 with this comment as a separate issue.

Reply:
The previous ticket [included the idea of throwing ConcurrentModificationException](https://github.com/apache/accumulo/pull/2717#issuecomment-1133596040) when the server detected that the properties changed. That would require the version information suggested in #2821 be incorporated into these changes.

Reply:
I missed that the first time around, I'll take a look at adding that version info to this PR this week and we can close out #2821 since it will be part of this instead.